MOTECH-2094: clean up scheduler API

kmadej opened the pull request:
This PR is connected with https://github.com/motech/motech/pull/403
I had to open new PR from my fork to push my changes. Generally this changes are very similar to related PR. I made only small adjustments.

kmadej commented:
test this please

pgesek commented on platform/event/src/main/java/org/motechproject/event/listener/proxy/EventAnnotationBeanPostProcessor.java:
```diff
@@ -20,8 +20,6 @@
 /**
  * Provides the <code>BeanPostProcessor</code> implementation for processing event annotations.
- *
- * @author yyonkov
```
I would leave the authors for old times sake

pgesek commented on modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/service/impl/MotechSchedulerServiceImpl.java:
```diff
@@ -192,41 +192,6 @@ public void safeScheduleRunOnceJob(RunOnceSchedulableJob schedulableJob) {
     }
 
     @Override
-    @Deprecated // nobody's using it and can't imagine a case for this
```
funny, aren't we updating jobs through UI now?

kmadej replied:
we are using updateJob(SchedulableJob job) for that.

pgesek commented on modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/service/SchedulerChannelProvider.java:
```diff
@@ -183,19 +181,12 @@ private String buildGetTriggerQuery(String subject) {
-        sb.append(" = \'");
-        sb.append(subject);
-        sb.append("\'");
+        sb.append("SELECT ").append(getCorrectName(JOB_NAME, isPostgres)).append(", ")
```
to be honest the previous version looked more clear, maybe a matter of preference

pgesek commented on modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/service/SchedulerChannelProvider.java:
```diff
@@ -160,18 +160,16 @@ private String buildGetTriggersQuery(int page, int pageSize) {
-        sb.append(pageSize);
-        sb.append(" OFFSET ");
-        sb.append((page - 1) * pageSize);
+        sb.append("SELECT ")
```
this one looks ok though

pgesek commented:
One more thing: 
1) Reorganize the exports in the pom alphabetically
2) Why do we have a star export for the service package?

kmadej commented:
I made few changes :

1. moved SchedulerChannelProvider from service to tasks package (because is connected with tasks module), then I had to export tasks package
2. moved MotechScheduledJob, MotechScheduler from service.impl package to service package, then I could remove star from service package export. (now impl is not exported)
3. moved scheduler task event parser name constanst to SchedulerConstanst class, because of package cycle between service and tasks package. Now MotechSchedulerJob doesn't import
constant from tasks package.
4. reorganized the exports in the pom alphabetically

pgesek commented:
@kmadej has conflicts

pgesek commented:
@kmadej still has conflicts
